hummingbot/strategy/pure_market_making/moving_price_band.py

```diff
@@ -0,0 +1,115 @@
+    :param price_ceiling_pct: reference price to set price band
+    :param price_band_refresh_time: reference price to set price band
+    '''
+    price_floor_pct: Decimal = -1
+    price_ceiling_pct: Decimal = 1
+    price_band_refresh_time: float = 86400
+    _price_floor: Decimal = 0
+    _price_ceiling: Decimal = 0
+    _set_time: float = 0
+
+    @property
+    def price_floor(self) -> Decimal:
+        '''get price floor'''
+        return self._price_floor
+
+    @property
+    def price_ceiling(self) -> Decimal:
+        '''get price ceiling'''
+        return self._price_ceiling
+
+    @property
+    def enabled(self) -> bool:
+        '''return state of moving price band'''
+        return True
+
+    def update(self, timestamp: float, price: Decimal) -> None:
+        """
+        Updates the price band.
+
+        :param timestamp: current timestamp of the strategy/connector
+        :param price: reference price to set price band
+        """
+        self._price_floor = (Decimal("100") + self.price_floor_pct) / Decimal("100") * price
+        self._price_ceiling = (Decimal("100") + self.price_ceiling_pct) / Decimal("100") * price
+        self._set_time = timestamp
+
+    def check_and_update_price_band(self, timestamp: float, price: Decimal) -> None:
+        '''
+        check if the timestamp has passed the defined refresh time before updating
+
+        :param timestamp: current timestamp of the strategy/connector
+        :param price: reference price to set price band
+        '''
+        if timestamp < self._set_time + self.price_band_refresh_time:
+            return
+        self.update(timestamp, price)
+
+    def check_price_floor_exceeded(self, price: Decimal) -> bool:
+        '''
+        check if the price has exceeded the price floor
+
+        :param price: price to check
+        '''
+        if price <= self.price_floor:
+            return True
+        return False
+
+    def check_price_ceiling_exceeded(self, price: Decimal) -> bool:
+        '''
+        check if the price has exceeded the price ceiling
+
+        :param price: price to check
+        '''
+        if price >= self.price_ceiling:
+            return True
+        return False
+
+    def switch(self, value: bool) -> "MovingPriceBand":
+        '''
+        switch between enabled and disabled state
+
+        :param value: set whether to enable or disable MovingPriceBand
+        '''
+        if value:
+            return MovingPriceBand(
+                price_floor_pct=self.price_floor_pct,
+                price_ceiling_pct=self.price_ceiling_pct,
+                price_band_refresh_time=self.price_band_refresh_time
+            )
+        return DisableMovingPriceBand(
+            price_floor_pct=self.price_floor_pct,
+            price_ceiling_pct=self.price_ceiling_pct,
+            price_band_refresh_time=self.price_band_refresh_time
+        )
+
+
+class DisableMovingPriceBand(MovingPriceBand):
```

Comment:
It might be better to call the class `DisabledMovingPriceBand`

Reply:
OK.